app/controllers/subscription_controller.rb

```diff
@@ -170,8 +171,8 @@ def multiple_add
         # user or node subscription
       end
     else
-      flash[:warning] = "You must be logged in to subscribe for email updates."
-      redirect_to "/login"
+      flash[:warning] = "You must be logged in to subscribe for email updates; please <a href='javascript:void()' onClick='login()'>log in</a> or <a href='/signup'>create an account</a>."
+      redirect_to "subscribe/multiple/" + params[:type] + params[:tagnames]
```

Comment:
redirect_to "subscribe/multiple/" + params[:type] + params[:tagnames]
will not work. 
Try following
You are not logged in. Go to `localhost:3000/subscribe/multiple/tag/kites,oil,,kite`.
If you succeed please give me a screenshot.
You will not succeed because the subscription page is only available to the already logged in person. Does this makes sense?

Reply:
Yes it does make sense, even I thought the original wouldn't work because it ideally should give an error. I didn't worry about this too much because I was showing the login modal for not signed in user so this condition wouldn't have arrived anyways.

Reply:
oh okay, I didn't know that, please tell me what I should do then.

Reply:
Change it to the previous state which I did. Then it will work fine

Reply:
Using return_to will be best here. After the person logs in they will get redirected to the required route.

Reply:
Could you please highlight the previous one? is it the /login one?

Reply:
Yeah... Just add return_to.